mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -280,6 +280,14 @@ void initialize(boolean isSubmit) throws Exception {
         LOGGER.info("Job {} initialized", this.jobId);
     }
 
+    private long getWorkerTimeoutSecs() {
+        if (mantisJobMetaData.getWorkerTimeoutSecs() > 0) {
+            return mantisJobMetaData.getWorkerTimeoutSecs();
+        } else {
+            return ConfigurationProvider.getConfig().getWorkerTimeoutSecs();
+        }
```

Comment:
I would move this logic to the MantisJobMetadataImpl. That way, everyone sees a consistent value of the WorkerTimeoutSecs.